CMakeLists.txt

```diff
@@ -10,6 +10,13 @@ include(cmake/StandardProjectSettings.cmake)
 # SET(CUDA_SEPARABLE_COMPILATION ON)
 # cuda_select_nvcc_arch_flags(ARCH_FLAGS)
 
+find_package(OpenMP)
+if (OPENMP_FOUND)
+    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
+    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+    set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")
+endif()
```

Comment:
Please change to modern CMake syntax
https://cliutils.gitlab.io/modern-cmake/chapters/packages/OpenMP.html